src/core/src/package_managers/TdnfPackageManager.py

```diff
@@ -0,0 +1,458 @@
+
+    def refresh_repo(self):
+        self.composite_logger.log("[TDNF] Refreshing local repo...")
+        self.invoke_package_manager(self.cmd_repo_refresh_template)
+
+    # region Get Available Updates
+    def invoke_package_manager_advanced(self, command, raise_on_exception=True):
+        """Get missing updates using the command input"""
+        self.composite_logger.log_verbose("[TDNF] Invoking package manager. [Command={0}]".format(str(command)))
+        code, out = self.env_layer.run_command_output(command, False, False)
+
+        if code not in [self.tdnf_exitcode_ok, self.tdnf_exitcode_no_applicable_packages, self.tdnf_exitcode_updates_available]\
+                and (command.startswith(self.single_package_upgrade_simulation_cmd) and code != self.tdnf_exitcode_update_no_action_err):
+            self.composite_logger.log_warning('[ERROR] Customer environment error. [Command={0}][Code={1}][Output={2}]'.format(command, str(code), str(out)))
+            error_msg = "Customer environment error: Investigate and resolve unexpected return code ({0}) from package manager on command: {1}".format(str(code), command)
+            self.status_handler.add_error_to_status(error_msg, Constants.PatchOperationErrorCodes.PACKAGE_MANAGER_FAILURE)
+            if raise_on_exception:
+                raise Exception(error_msg, "[{0}]".format(Constants.ERROR_ADDED_TO_STATUS))
+            # more return codes should be added as appropriate
+        else:  # verbose diagnostic log
+            self.composite_logger.log_debug('[TDNF] Invoked package manager. [Command={0}][Code={1}][Output={2}]'.format(command, str(code), str(out)))
+        return out, code
+
+    # region Classification-based (incl. All) update check
+    def get_all_updates(self, cached=False):
+        """Get all missing updates"""
+        self.composite_logger.log_verbose("[TDNF] Discovering all packages...")
+        if cached and not len(self.all_updates_cached) == 0:
+            self.composite_logger.log_debug("[TDNF] Get all updates : [Cached={0}][PackagesCount={1}]]".format(str(cached), len(self.all_updates_cached)))
+            return self.all_updates_cached, self.all_update_versions_cached  # allows for high performance reuse in areas of the code explicitly aware of the cache
+
+        out = self.invoke_package_manager(self.tdnf_check)
+        self.all_updates_cached, self.all_update_versions_cached = self.extract_packages_and_versions(out)
+
+        self.composite_logger.log_debug("[TDNF] Get all updates : [Cached={0}][PackagesCount={1}]]".format(str(False), len(self.all_updates_cached)))
+        return self.all_updates_cached, self.all_update_versions_cached
+
+    def get_security_updates(self):
+        """Get missing security updates"""
+        self.composite_logger.log_verbose("[TDNF] Discovering 'security' packages...")
+        self.composite_logger.log_warning("Classification based categorization of patches is not available in Azure Linux as of now.")
```

Comment:
Don't do this here as this line will be invoked many times in an execution and flood the error list with repeats of this.

Reply:
done